### PR TITLE
Fix an error when compiling with ACL: conversion from gsl::span<const long int> to non-scalar type std::vector<long int>

### PR DESCRIPTION
**Pull request by fengyuentau**
**Description**: Same as https://github.com/microsoft/onnxruntime/issues/9893. Credit to the author of the issue.

**Motivation and Context**
- Why is this change required? What problem does it solve?
  - It fixes the error when compiling with ACL: `conversion from gsl::span<const long int> to non-scalar type std::vector<long int>`
- If it fixes an open issue, please link to the issue here.
  - https://github.com/microsoft/onnxruntime/issues/9893

### CODE REVIEW

File: onnxruntime/core/providers/acl/tensor/concat.cc [hunk 1]
```diff
@@ -32,7 +32,7 @@ Status Concat<T>::Compute(OpKernelContext* ctx) const {
     input_tensors.push_back(ctx->Input<Tensor>(i));
   }
 
-  std::vector<int64_t> output_dims = input_tensors[0]->Shape().GetDims();
+  std::vector<int64_t> output_dims = input_tensors[0]->Shape()..GetDimsAsVector();
```
Comment (edgchen1) [inline onnxruntime/core/providers/acl/tensor/concat.cc:35]:
```suggestion
  std::vector<int64_t> output_dims = input_tensors[0]->Shape().GetDimsAsVector();
```

**Comment by fengyuentau** [comment]
@edgchen1  Thanks for the correction. A classic copy and paste typo.

File: onnxruntime/core/providers/acl/tensor/concat.cc [hunk 2]
```diff
@@ -32,7 +32,7 @@ Status Concat<T>::Compute(OpKernelContext* ctx) const {
     input_tensors.push_back(ctx->Input<Tensor>(i));
   }
 
-  std::vector<int64_t> output_dims = input_tensors[0]->Shape().GetDims();
+  std::vector<int64_t> output_dims = input_tensors[0]->Shape().GetDimsAsVector();
```
Comment (yuslepukhin) [inline onnxruntime/core/providers/acl/tensor/concat.cc:35]: >GetDimsAsVector [](http://example.com/codeflow?start=63&length=15)

This function no longer exists. See coding standards. All shapes are now handled with TensorShapeVector which has small buffer optimization

Comment (fengyuentau) [reply onnxruntime/core/providers/acl/tensor/concat.cc:35]: Well, I was using v1.10.0 and testing in the version of code, in which `GetDimsAsVector` exists but `AsShapeVector` does not. I will change to `AsShapeVector` as suggested.

Comment (yuslepukhin) [reply onnxruntime/core/providers/acl/tensor/concat.cc]: The function does not return std::vector. Please, read the comment above.

Comment (yuslepukhin) [reply onnxruntime/core/providers/acl/tensor/concat.cc]: Just replace `std::vector<int64_t>` with `auto`, should work.

**Comment by yuslepukhin** [comment]
See https://github.com/microsoft/onnxruntime/pull/10667

**Pull request closed without merging**